crates/rover-client/src/operations/graph/check_workflow/runner.rs

```diff
@@ -42,26 +42,27 @@ pub fn run(
     client: &StudioClient,
 ) -> Result<CheckWorkflowResponse, RoverClientError> {
     let graph_ref = input.graph_ref.clone();
-    let mut data;
+    let mut url: Option<String> = None;
     let now = Instant::now();
     loop {
-        data = client.post::<GraphCheckWorkflowQuery>(input.clone().into())?;
-        let graph = data.clone().graph.ok_or(RoverClientError::GraphNotFound {
-            graph_ref: graph_ref.clone(),
-        })?;
-        if let Some(check_workflow) = graph.check_workflow {
-            if !matches!(check_workflow.status, CheckWorkflowStatus::PENDING) {
-                break;
+        let result = client.post::<GraphCheckWorkflowQuery>(input.clone().into());
+        if result.is_ok() {
+            let data = result?;
+            let graph = data.clone().graph.ok_or(RoverClientError::GraphNotFound {
+                graph_ref: graph_ref.clone(),
+            })?;
+            if let Some(check_workflow) = graph.check_workflow {
+                if !matches!(check_workflow.status, CheckWorkflowStatus::PENDING) {
+                    return get_check_response_from_data(data, graph_ref);
+                }
             }
+            url = get_target_url_from_data(data);
         }
         if now.elapsed() > Duration::from_secs(input.checks_timeout_seconds) {
-            return Err(RoverClientError::ChecksTimeoutError {
-                url: get_target_url_from_data(data),
-            });
+            return Err(RoverClientError::ChecksTimeoutError { url });
```

Comment:
It's probably a good idea to store the latest error result (if there was one) and display that alongside the timeout rather than just saying it timed out. A timeout with nothing but successful requests is a bit different than a timeout with nothing but errored requests.

Reply:
It might even be good to log out the errors and say "this error may be transient, we are retrying the request" instead of doing nothing